framework/src/mfem/utils/CoefficientManager.C

```diff
@@ -0,0 +1,169 @@
+#ifdef MFEM_ENABLED
+
+#include "CoefficientManager.h"
+#include <algorithm>
+
+namespace Moose::MFEM
+{
+
+mfem::Coefficient &
+CoefficientManager::declareScalar(const std::string & name, std::shared_ptr<mfem::Coefficient> coef)
+{
+  this->_scalar_coeffs.addCoefficient(name, coef);
+  return *coef;
+}
+
+mfem::Coefficient &
+CoefficientManager::declareScalar(const std::string & name, const std::string & existing_coef)
+{
+  return this->declareScalar(name, this->_scalar_coeffs.getCoefficientPtr(existing_coef));
+}
+
+mfem::Coefficient &
+CoefficientManager::declareScalarProperty(const std::string & name,
+                                          const std::vector<std::string> & blocks,
+                                          std::shared_ptr<mfem::Coefficient> coef)
+{
+  this->_scalar_coeffs.addPiecewiseBlocks(name, coef, blocks);
+  return getScalarCoefficient(name);
+}
+
+mfem::Coefficient &
+CoefficientManager::declareScalarProperty(const std::string & name,
+                                          const std::vector<std::string> & blocks,
+                                          const std::string & existing_coef)
+{
+  std::shared_ptr<mfem::Coefficient> coef = this->_scalar_coeffs.getCoefficientPtr(existing_coef);
+  if (std::dynamic_pointer_cast<mfem::PWCoefficient>(coef))
+  {
+    throw MooseException(
+        "Properties must not be defined out of other properties or piecewise coefficients.");
+  }
```

Comment:
```suggestion
    throw MooseException(
        "Properties must not be defined out of other properties or piecewise coefficients.");
```

Reply:
I guess maybe these should use `mooseError()` too?